src/components/templates/PartyMap/components/ScheduleItem/ScheduleItem.tsx

```diff
@@ -26,40 +27,47 @@ export const ScheduleItem: React.FunctionComponent<PropsType> = ({
   roomUrl,
 }) => {
   const dispatch = useDispatch();
+
+  const schedulePrimaryClasses = classNames({
+    "ScheduleItem--primary": isCurrentEvent,
+  });
+
+  const scheduleItemTimeSectionClasses = classNames(
+    "ScheduleItem__time-section",
+    schedulePrimaryClasses
+  );
+
   return (
-    <div className="schedule-item-container">
-      <div className={`time-section ${isCurrentEvent ? "primary" : ""}`}>
-        <div>
-          <b>{formatUtcSeconds(event.start_utc_seconds)}</b>
-        </div>
-        <div>
+    <div className="ScheduleItem">
+      <div className={scheduleItemTimeSectionClasses}>
+        <span className="ScheduleItem__event-date">
+          {labelDayRoomSchedule(event.start_utc_seconds)}
+        </span>
+        <span className="ScheduleItem__event-time">
+          {formatUtcSeconds(event.start_utc_seconds)}
+        </span>
+        <span className="ScheduleItem__event-time">
           {formatUtcSeconds(
             event.start_utc_seconds + event.duration_minutes * 60
           )}
-        </div>
+        </span>
       </div>
```

Comment:
It would be better if we could use `eventStartTime` / `eventEndTime` from `utils/event` here rather than doing the calculations all over again.

`formatUtcSeconds` is defined as follows:

```typescript
export function formatUtcSeconds(utcSeconds?: number | null) {
  return utcSeconds ? format(fromUnixTime(utcSeconds), "p") : "(unknown)";
}
```

The main part of this is the `format` with pattern `p` (Long localized time eg. `12:00 AM`):

- https://date-fns.org/v2.21.3/docs/format

We already have `formatTimestampToDisplayHoursMinutes` which uses the pattern `h:mm aa`

- Hour [1-12]: `h`, eg. `1, 2, ..., 11, 12`
- Minute: `mm`, eg. `00, 01, ..., 59`
- AM, PM: `a..aa`, eg. `AM, PM`

This currently accepts `timestamp` (`number`) as it's param, but since it's just passing directly to `format`, the type can be changed to accept `Date | number`.

That would then allow the return value from `eventStartTime` / `eventEndTime` to be passed directly to it.

---

`formatTimestampToDisplayHoursMinutes` is currently only used in `src/components/atoms/ChatMessageInfo/ChatMessageInfo.tsx` and would be fine if it used the `p` format.

Reply:
Refactoring, it would end up looking like (this is directly copy/pastable):

```typescript
/**
 * Format UTC seconds as a string representing the time in long localized time format (eg. 12:00 AM).
 *
 * @example
 *   formatTimestampToDisplayHoursMinutes(1618509600)
 *   // '9:35 AM'
 *
 * @param date
 *
 * @see https://date-fns.org/docs/format
 */
export const formatTimestampToDisplayHoursMinutes = (date: Date | number) =>
  format(date, "p");
```

And then we can use the following here:

```typescript
 <span className="ScheduleItem__event-time">
  {formatTimestampToDisplayHoursMinutes(eventStartTime(event))}
</span>
<span className="ScheduleItem__event-time">
  {formatTimestampToDisplayHoursMinutes(eventEndTime(event))}
</span>
```

Reply:
In `src/components/organisms/NavBarSchedule/NavBarSchedule.tsx` we have `formatDayLabel` defined inline, which uses:

```typescript
const formatDayLabel = (day: Date | number) => {
  if (isScheduleTimeshifted) {
    return format(day, "E, LLL d");
  } else {
    return isToday(day) ? "Today" : format(day, "E");
  }
};
```

`labelDayRoomSchedule` in `utils/time` in this PR is defined as:

```typescript
export const labelDayRoomSchedule = (startUtcSeconds: number) => {
  const date = fromUnixTime(startUtcSeconds);

  if (isToday(date)) return "";
  if (isTomorrow(date)) return "Tomorrow";

  return formatDate(startUtcSeconds);
};
```

`formatDate` in `utils/time` is defined as:

```typescript
export function formatDate(utcSeconds: number) {
  return format(fromUnixTime(utcSeconds), "MMM do");
}
```

---

Both of these are sort of different variations on `formatRelative`, but only for dates rather than times:

- https://date-fns.org/v2.22.1/docs/formatRelative

Looking at `format`:

- https://date-fns.org/v2.22.1/docs/format

For these patterns:

- `E, LLL d`, eg. `Mon, Jun 7`
  - `E`: Day of week (formatting), eg. `Mon, Tue, Wed, ..., Sun`
  - `LLL`: Month (stand-alone), eg. `Jan, Feb, ..., Dec`
  - `d`: Day of month, eg. `1, 2, ..., 31`
- `MMM do`, eg. `Jan 7th`
  - `MMM`: Month (formatting), eg. `Jan, Feb, ..., Dec`
  - `do`: Day of month, eg. `1st, 2nd, ..., 31st`

Based on that:
- `LLL` and `MMM` appear to be equivalent.
- We could potentially replace `d` with `do` or vice versa quite easily

Reply:
- refactor formatTimestampToDisplayHoursMinutes to use format pattern '… …49a8bec
- refactor ScheduleItem to use eventStartTime / eventEndTime with forma… … 626cf09
- refactor formatDate in utils/time to accept dateOrTimestamp 7b7b63a
- replace labelDayRoomSchedule with formatDateRelativeToNow b58a244
- refactor formatDayLabel in NavBarSchedule to use formatDateRelativeToNow a825845

Reply:
Thanks! Done: 9bd0adfd98424ff18e366d672c9a30b86838ca37